networkx/algorithms/community/modularity.py

```diff
@@ -0,0 +1,93 @@
+# modularity.py - functions for computing modularity of a partition
+#
+# Copyright 2011 Ben Edwards <bedwards@cs.unm.edu>.
+# Copyright 2011 Aric Hagberg <hagberg@lanl.gov>.
+# Copyright 2015 NetworkX developers.
+#
+# This file is part of NetworkX.
+#
+# NetworkX is distributed under a BSD license; see LICENSE.txt for more
+# information.
+"""Functions for computing the modularity of a partition."""
+from __future__ import division
+
+from itertools import product
+
+import networkx as nx
+from .community_utils import is_partition
+
+
+def modularity(G, communities, weight='weight'):
```

Comment:
Of course you can.  But bad stuff is more likely to happen #2087 and another recent case I can't remember right now.